integration-tests/test_satellite.py

```diff
@@ -0,0 +1,113 @@
+import pytest
+import subprocess
+import os
+import yaml
+import traceback
+import contextlib
+from time import sleep
+
+
+pytestmark = pytest.mark.usefixtures("register_subman")
+
+
+def test_obfuscation(insights_client):
+    """
+    Test if the satellite hostname is obfuscated
+    Related to https://bugzilla.redhat.com/show_bug.cgi?id=1963917
+
+    """
+    # Register system to insights
+    insights_client.register()
+    assert insights_client.is_registered
+
+    # Record the current hostname
+    cmd = ["cat", "/etc/hostname"]
+    output = subprocess.check_output(cmd)
+    previous_hostname = output.decode("utf-8").strip()
+
+    try:
+        # Set a new hostname
+        hostname_to_test = "insights-client-satellite"
+        set_hostname(hostname_to_test)
+        cmd = ["hostnamectl", "status"]
+        output = subprocess.check_output(cmd)
+        assert hostname_to_test in output.decode("utf-8").strip()
+
+        # redaction on hostname:
+        data_redaction(hostname_to_test)
+
+        # check the collection, if redaction pattern is filtered out
+        # collect data
+        archive_result = insights_client.run("--no-upload")
+        assert archive_result.returncode == 0
+
+        # extract the collected tar file
+        archive_location = archive_result.stdout.split("\n")[-2].split(" ")[-1]
+        cmd = ["tar", "-xzvf", archive_location]
+        subprocess.check_call(cmd)
+
+        # the redaction data should not locate in the collected files
+        archive_extract = archive_location.split(".")[0].split("/")[-1]
+        cmd = "cd %s; grep -rnw . -e %s" % (archive_extract, hostname_to_test)
+        # should not find the obfuscate pattern, the return code should not zero.
+        returncode = subprocess.check_call(cmd, shell=True)
+        assert returncode
+    except Exception as e:
+        traceback.print_exc()
+    finally:
+        # Recover the previous hostname for the system
+        set_hostname(previous_hostname)
+        # Remove all data redaction
+        cmd = ["rm", "-f", "/etc/insights-client/file-content-redaction.yaml"]
+        subprocess.check_call(cmd)
+
+
+def test_set_ansible_host_info(insights_client):
+    """
+    Test if the ansible-host can be set with satellite
+    Related to https://issues.redhat.com/browse/RHELPLAN-160348,
+    and https://bugzilla.redhat.com/show_bug.cgi?id=222729
+    """
+    # Register system against Satellite, and register insights through satellite
+    insights_client.register()
+    sleep(10)
+    assert insights_client.is_registered
+
+    # Update ansible-host
+    set_ansible_host = insights_client.run("--ansible-host='foo.example.com'")
+    assert set_ansible_host.returncode == 0
+
+
+def test_register_group_option(insights_client):
```

Comment:
this test seems related to the registration, and thus IMHO ought to better be in `test_register.py`; IMHO the fact that it fails on Satellite does not make it "specific" to Satellite, and running on Satellite seems simply a different scenario/setup for the test

Reply:
Jan is focusing the test_register.py migration(https://issues.redhat.com/browse/CCT-340). We can move this case when his PR is merged. What's your thoughts? @Archana-PandeyM @jstavel

Reply:
other test cases can be added to an existing file, not an issue IMHO :)

Reply:
When we have a solution for insights_client.run("--register", "--group=tag"), I will move this case.